internal/gateway/retrieve.go

```diff
@@ -58,26 +113,45 @@ func (c *Client) Retrieve() *cobra.Command {
 
 			gways := gwayList.GetMetalGateways()
 
-			data := make([][]string, len(gways))
-			metalGways := make([]*metal.MetalGateway, len(gways))
+			data := make([][]string, 0, len(gways))
+			var gateways []gatewayIsh
+
+			for _, gwaysInner := range gways {
+				var gway gatewayIsh
+				if gwaysInner.MetalGateway != nil {
```

Comment:
IMO introducing additional types/interfaces isn't worth it if we're still going to end up with an if block inside a for loop.

Reply:
I moved this out of the loop